spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageConverterConfigurer.java

```diff
@@ -83,25 +88,35 @@
 	private final BindingServiceProperties bindingServiceProperties;
 
 	private ConfigurableListableBeanFactory beanFactory;
+	
+	private final Map<String, PartitionKeyExtractorStrategy> partitionKeyExtractors;
+	
+	private final Map<String, PartitionSelectorStrategy> partitionSelectors;
 
+	@SuppressWarnings("unchecked")
 	public MessageConverterConfigurer(BindingServiceProperties bindingServiceProperties,
 			CompositeMessageConverterFactory compositeMessageConverterFactory) {
+		this(bindingServiceProperties, compositeMessageConverterFactory, Collections.EMPTY_MAP, Collections.EMPTY_MAP);
+	}
+	
+	@SuppressWarnings("unchecked")
+	public MessageConverterConfigurer(BindingServiceProperties bindingServiceProperties,
+			CompositeMessageConverterFactory compositeMessageConverterFactory,
+			Map<String, PartitionKeyExtractorStrategy> partitionKeyExtractors,
+			Map<String, PartitionSelectorStrategy> partitionSelectors) {
 		Assert.notNull(compositeMessageConverterFactory,
 				"The message converter factory cannot be null");
 		this.bindingServiceProperties = bindingServiceProperties;
 		this.compositeMessageConverterFactory = compositeMessageConverterFactory;
+		this.partitionKeyExtractors = partitionKeyExtractors == null ? Collections.EMPTY_MAP : partitionKeyExtractors;
```

Comment:
Remove the `@SuppressWarnings` and use the type-safe variant - `Collections.emptyMap()`.